### DISCUSSION TITLE
Callback param types on `messageReactionAdd` not working

### DISCUSSION THREAD

**Discussion by SayakMukhopadhyay** [discordjs/discord.js#6698] [Q&A] [answered]
I am having a weird issue while developing in typescript. I am trying to listen to the `messageReactionAdd` event and want to type the listener function parameters. Example code
```ts
this.client.on('messageReactionAdd', (reaction: MessageReaction, user: User) => {
  //code
});
```
Typescript doesn't like it and says "No overload matches this call". 
![image](https://user-images.githubusercontent.com/13748882/134822946-63e62791-562c-4556-8556-c493ae5bbf65.png)

If I change the `moduleResolution` type to `classic` from `node`, this error doesn't appear anymore. Moreover, this error didnt happen with the same code in v12. This is the difference in the same definition in 12 vs 13
![image](https://user-images.githubusercontent.com/13748882/134823061-cfe8ba16-cef2-40de-83f7-be5495285aa5.png)
Seems like the typings were changed. So, if I manually edit the typing to change it back to v12 style, things work as expected.
What's going wrong? Any ideas?

**Comment by ImRodry** [comment]
You shouldn’t cast the type of reaction and user, they will be casted automatically. Along with that make sure you’re not returning anything in the event, you must always return void

**Reply by SayakMukhopadhyay** [reply]
Thing is in webstorm, if I just do
```ts
this.client.on('messageReactionAdd', (reaction, user) => {
  reaction.message
  user.client
});
```

The type of `reaction` is getting shown as `any`
![image](https://user-images.githubusercontent.com/13748882/134823969-d9b9ab81-8622-45dd-9fdd-40b53ed34cc6.png)
The type hints still show the correct hints and it works without a hitch in VSCode. Even doing
```ts
this.client.on('messageReactionAdd', (reaction: MessageReaction, user: User) => {
  //code
});
```
works fine in VSCode.

**Reply by ImRodry** [reply]
You should really be using VSCode for TypeScript, it's the most optimized IE

**Reply by monbrey** [reply, marked as answer]
Definitely seems to be Webstorm at fault here, if you look at the error it appears to be inserting additional `any` args before each of the real ones.

![image](https://user-images.githubusercontent.com/5294381/134826818-ed374919-8805-4a07-a6ef-7adc2ee4ef19.png)

args_0 and args_2 shouldn't exist - no idea where they're coming from. What Webstorm seems to think are args_1 and args_3 are the real ones, which also seems to align with the typings you're seeing on hover.

**Reply by SayakMukhopadhyay** [reply]
Yeah. I think I will create a small reproduction and raise it with JB. Thanks for the insights!

**Reply by SayakMukhopadhyay** [reply]
I just tried building my project and it failed. I can reproduce this issue. See https://codesandbox.io/s/discord-js-typings-bug-8zif7. If you want to run commands, please fork the sandbox and create a new terminal to run the build script. Here is what I got
![image](https://user-images.githubusercontent.com/13748882/134935202-eb5fc614-a6d9-4319-9dec-a11c5a439ad9.png)
